models/Document.cfc

```diff
@@ -46,6 +46,10 @@ component accessors="true" {
 	 */
 	property name="params";
 
+	/**
+	 * Any evaluated script field results
+	 */
+	property name="scriptFields" type="struct";
```

Comment:
If we are going to implement this for script fields, let's add handling for runtime fields at the same time ( https://www.elastic.co/guide/en/elasticsearch/reference/current/runtime-retrieving-fields.html ).

For both, I feel like we need `getMemento` to include those - or at least allow an argument to flatten them together in to a single memento.

Reply:
You marked this as resolved, but I don't see the runtime fields support added anywhere.

Reply:
Elasticsearch is confusing in that script fields are `"script_fields"`, but runtime fields are just `"fields"`. :facepalm: But you're right, I missed adding a `addField()` appender method for runtime/a la carte document fields to return.

The runtime field support on the response side is pretty straightforward, and is retrieved the same as script fields.

From the docs:

---

In a similar fashion, runtime fields can also be fetched and will appear in the document's `fields` struct. This example retrieves the `"fuel_usage_in_mpg"` runtime field as well as the indexed `"make"` and `"model"` fields:

```js
var hits = searchBuilder.new( "itinerary" )
             .setFields( [ "fuel_usage_in_mpg", "make", "model" ] )
             .execute()
             .getHits();
for( hit in hits ){
    var result = hit.getFields();
    writeOutput( "This #result.make# #result.model# gets #fuel_mpg#/gallon" );
}
```

To access document `fields` as well as the `_source` properties, use`hit.getDocument( includeFields = true)`:

```js
var result = searchBuilder.execute();
for( hit in result.getHits() ){
    var document = document.getDocuments( includeFields = true );
    writeOutput( "This #document.make# #document.model# gets #fuel_mpg#/gallon" );
}
```